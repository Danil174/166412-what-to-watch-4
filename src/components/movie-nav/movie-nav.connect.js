import {connect} from "react-redux";
import {ActionCreator} from "../../reducer/app-state/app-state.js";
import {getActiveMovieTab} from "../../reducer/app-state/selectors.js";
import MovieNav from "./movie-nav.jsx";

const mapStateToProps = (state) => ({
  activeTab: getActiveMovieTab(state)
});

const mapDispatchToProps = (dispatch) => ({
  onTabItemClick(tab) {
    dispatch(ActionCreator.changeMovieTab(tab));
  }
});

export {MovieNav};
export default connect(mapStateToProps, mapDispatchToProps)(MovieNav);
