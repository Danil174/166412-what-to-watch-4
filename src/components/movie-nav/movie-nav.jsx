import React from "react";
import {connect} from "react-redux";
import {ActionCreator} from "../../reducer/app-state/app-state.js";
import {getActiveMovieTab} from "../../reducer/app-state/selectors.js";
import {checkNavTagretClick} from "../../utils/common.js";
import PropTypes from "prop-types";

const renderTab = (tab, activeTab) => {
  const activeClass = activeTab === tab ? `movie-nav__item--active` : 0;
  const tabClass = `movie-nav__item ${activeClass}`;

  return (
    <li
      key={tab}
      className={tabClass}>
      <a
        className="movie-nav__link"
        data-item={tab}
      >
        {tab}
      </a>
    </li>
  );
};
const MovieNav = (props) => {
  const {tabs, activeTab, onTabItemClick} = props;
  return (
    <nav
      className="movie-nav movie-card__nav"
      onClick={(evt) => {
        evt.preventDefault();
        const tab = checkNavTagretClick(evt.target, activeTab);
        onTabItemClick(tab);
      }}
    >
      <ul className="movie-nav__list">
        {tabs.map((tab) => renderTab(tab, activeTab))}
      </ul>
    </nav>
  );
};

MovieNav.propTypes = {
  onTabItemClick: PropTypes.func.isRequired,
  tabs: PropTypes.arrayOf(PropTypes.string).isRequired,
  activeTab: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => ({
  activeTab: getActiveMovieTab(state)
});

const mapDispatchToProps = (dispatch) => ({
  onTabItemClick(tab) {
    dispatch(ActionCreator.changeMovieTab(tab));
  }
});

export {MovieNav};
export default connect(mapStateToProps, mapDispatchToProps)(MovieNav);
