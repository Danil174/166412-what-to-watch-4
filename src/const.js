export const TextMovieRatings = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`,
};

export const SmallCardVideoSettings = {
  WIDTH: `100%`,
  HEIGHT: `100%`,
  IS_MUTED: true,
};

export const DEFAULT_GENRE = `All genres`;
export const MAX_GENRES_LENGTH = 9;
export const INITTIAL_LIST_LENGTH = 8;
export const SIMILAR_LIST_LENGTH = 4;

export const AuthorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`,
};

export const smallCardVideoStartTimeOut = 1000;

export const LoadErrorsTexts = {
  LIST_FAIL: `Не удалось загрузить список фильмов.`,
  PROMO_FAIL: `Не удалось загрузить промо фильм.`,
};

export const AppRoute = {
  LOGIN: `/login`,
  MOVIE_PAGE: `/movie_page`,
  FAVORITES: `/favorites`,
  ROOT: `/`,
};
