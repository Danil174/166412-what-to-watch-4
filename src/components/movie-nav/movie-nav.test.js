import React from "react";
import renderer from "react-test-renderer";
import {MovieNav} from "./movie-nav.jsx";

import {Provider} from "react-redux";
import configureStore from "redux-mock-store";
import NameSpace from "../../reducer/name-space.js";
import {MovieTabsMap, MovieTabs} from "../../const.js";

const mockStore = configureStore([]);

describe(`MovieNavTest`, () => {
  it(`render MovieNav`, () => {
    const store = mockStore({
      [NameSpace.APP_STATE]: {
        activeMovieTab: MovieTabsMap.OVERVIEW,
      },
    });
    const tree = renderer
      .create(
          <Provider store={store}>
            <MovieNav
              activeTab={MovieTabsMap.OVERVIEW}
              tabs={MovieTabs}
              onTabItemClick={()=>{}}
            />
          </Provider>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
