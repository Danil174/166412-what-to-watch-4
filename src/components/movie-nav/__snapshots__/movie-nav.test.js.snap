// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieNavTest render MovieNav 1`] = `
<nav
  className="movie-nav movie-card__nav"
  onClick={[Function]}
>
  <ul
    className="movie-nav__list"
  >
    <li
      className="movie-nav__item movie-nav__item--active"
    >
      <a
        className="movie-nav__link"
        data-item="Overview"
      >
        Overview
      </a>
    </li>
    <li
      className="movie-nav__item 0"
    >
      <a
        className="movie-nav__link"
        data-item="Details"
      >
        Details
      </a>
    </li>
    <li
      className="movie-nav__item 0"
    >
      <a
        className="movie-nav__link"
        data-item="Reviews"
      >
        Reviews
      </a>
    </li>
  </ul>
</nav>
`;
